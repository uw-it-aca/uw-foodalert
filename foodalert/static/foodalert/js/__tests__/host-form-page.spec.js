"use strict";
import Vue from 'vue';
import BootstrapVue from 'bootstrap-vue'
Vue.use(BootstrapVue);

import {mount} from '@vue/test-utils';
import FormPage from '../pages/host/form-page.vue';

import axios from 'axios'

jest.mock('axios', () => ({
	get: jest.fn(),
	post: jest.fn()
}))

const $routes = {
    push: jest.fn()
}

describe('Food Categories Page', function () {
	test('renders correctly', function () {
		axios.get.mockImplementation(() => Promise.resolve({data: []}));
		axios.post.mockImplementation(() => Promise.resolve({data: [{name: "allergne_1"}, {name: "allergen_2"}]}));
		var wrapper = mount(FormPage, {
		});
		expect(wrapper.element).toMatchSnapshot();
	});

	test('redirect to update', function () {
		axios.get.mockImplementation(() => Promise.resolve({data: [{ended: true}]}));
		axios.post.mockImplementation(() => Promise.resolve({data: []}));
		var wrapper = mount(FormPage, {
		});
		expect(wrapper.element).toMatchSnapshot();
	});

	test('form submit', function () {
		jest.clearAllMocks()
		axios.get.mockImplementation(() => Promise.resolve({data: []}));
		axios.post.mockImplementation(() => Promise.resolve({data: []}));
	    var wrapper = mount(FormPage, {
	        $routes
	    });
	    var inputs = wrapper.findAll('input')
	    inputs.at(0).setValue("some random value")
	    inputs.at(1).setValue("some random value")
	    inputs.at(2).setValue("12:00 AM")
	    inputs.at(3).setValue("some random value")

	    var textarea = wrapper.find('textarea')
	    textarea.setValue("some random value")

		var form = wrapper.find("form")
		form.trigger("submit")

	    var buttons = wrapper.findAll('button')
	    // buttons.at(0).trigger('click')
		buttons.at(4).trigger('click.native')
	});

	test('reset test', function () {
		axios.get.mockImplementation(() => Promise.resolve({data: []}));
		axios.post.mockImplementation(() => Promise.resolve({data: []}));
	    var wrapper = mount(FormPage, {
	        $routes
	    });

	    var buttons = wrapper.findAll('button')
	    buttons.at(1).trigger('click')
	});
})
