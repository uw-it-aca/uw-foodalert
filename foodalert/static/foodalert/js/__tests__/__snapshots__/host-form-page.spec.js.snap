// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Food Categories Page redirect to update 1`] = `
<div
  class="page"
>
  <div
    id="standard-notification"
  >
    <div
      class="collapse"
      id="notif-container"
      style="display: none;"
    >
      <div
        class="text-center py-3 container-fluid"
      >
        <div
          class="container"
        >
          <div
            class="row justify-content-center"
          >
            <div
              class="col-md-8"
            >
              <div
                class="h5 text-white pt-2"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="page-content pb-2"
  >
     
    <div
      class="container"
    >
      <div
        class="row justify-content-center"
      >
        <div
          class="page-content-padding col-md-8"
        >
          <h1
            class="pt-4 pb-2"
            id="standard-heading"
          >
            
        Compose notification
    
          </h1>
           
          <div
            id="standard-body"
          >
            <div>
              <div
                id="submitconfirmation___BV_modal_outer_"
                style="position: absolute; z-index: 1040;"
              >
                <div
                  aria-hidden="true"
                  class="modal fade"
                  id="submitconfirmation"
                  role="dialog"
                  style="display: none;"
                  tabindex="-1"
                >
                  <div
                    class="modal-dialog modal-md"
                  >
                    <div
                      aria-describedby="submitconfirmation___BV_modal_body_"
                      aria-labelledby="submitconfirmation___BV_modal_header_"
                      class="modal-content"
                      id="submitconfirmation___BV_modal_content_"
                      role="document"
                    >
                      <header
                        class="modal-header"
                        id="submitconfirmation___BV_modal_header_"
                      >
                        <h5
                          class="modal-title"
                        >
                          Confirmation
                        </h5>
                        <button
                          aria-label="Close"
                          class="close"
                          type="button"
                        >
                          Ã—
                        </button>
                      </header>
                      <div
                        class="modal-body"
                        id="submitconfirmation___BV_modal_body_"
                      >
                        <p>
                          
                We will send your notification to Hungry Husky Subscribers.
            
                        </p>
                         
                        <div
                          class="custom-preview-box"
                        >
                          <span>
                            Hot Indian buffet food
                          </span>
                           from
                
                          <span>
                            FIUTS weekly club meeting
                          </span>
                          .
                
                          <br />
                           
                          <br />
                          
                Quantity: 
                          <span>
                            About 8 full meals
                          </span>
                           
                          <br />
                          
                End time:
                
                          <span>
                            --:-- --
                          </span>
                           
                          <br />
                          
                Location: 
                          <span>
                            HUB 130
                          </span>
                           
                          <!---->
                           
                          <br />
                           
                          <!---->
                        </div>
                      </div>
                      <footer
                        class="modal-footer"
                        id="submitconfirmation___BV_modal_footer_"
                      >
                        <button
                          class="btn btn-outline-secondary"
                          type="button"
                        >
                          Cancel
                        </button>
                        <button
                          class="btn btn-primary"
                          type="button"
                        >
                          Send
                        </button>
                      </footer>
                    </div>
                  </div>
                </div>
                <!---->
                <!---->
              </div>
            </div>
             
            <form
              class=""
            >
              <label
                class="mt-2 standard-label"
                for="event-name"
              >
                Event name
              </label>
               
              <input
                aria-describedby="Name of the event"
                aria-required="true"
                class="mb-3 standard-placeholder form-control"
                id="event-name"
                placeholder="FIUTS weekly club meeting"
                required="required"
                type="text"
              />
               
              <label
                class="standard-label"
                for="food-description"
              >
                Describe the food
              </label>
               
              <textarea
                aria-describedby="Describe the food"
                aria-required="true"
                class="mb-3 standard-placeholder form-control"
                id="food-description"
                placeholder="Hot Indian buffet food"
                required="required"
                rows="2"
                wrap="soft"
              />
               
              <label
                class="standard-label"
                for="quantity"
              >
                Quantity
              </label>
               
              <input
                aria-describedby="Quantity of food"
                aria-required="true"
                class="mb-3 standard-placeholder form-control"
                id="quantity"
                placeholder="About 8 full meals"
                required="required"
                type="text"
              />
               
              <label
                class="standard-label"
                for="end-time"
              >
                End time (when food service will end)
              </label>
               
              <input
                aria-describedby="End time of the event"
                aria-required="true"
                class="mb-3 standard-placeholder form-control"
                id="end-time"
                required="required"
                type="time"
              />
               
              <label
                class="standard-label"
                for="location"
              >
                Location
              </label>
               
              <input
                aria-describedby="Location of the event"
                aria-required="true"
                class="mb-4 standard-placeholder form-control"
                id="location"
                placeholder="HUB 130"
                required="required"
                type="text"
              />
               
              <h2
                class="mt-4"
              >
                Food specifications
              </h2>
               
              <h5
                class="standard-label"
              >
                Does the food contain allergens?
              </h5>
               
              <p>
                It's ok if you are unsure, just select to the best of your knowledge.
              </p>
               
              <div
                class="px-0 container"
              >
                <div
                  class=""
                  id="allergens-checkbox"
                  role="group"
                  tabindex="-1"
                >
                  <div
                    class="row"
                  />
                </div>
              </div>
               
              <h5
                class="mt-4 standard-label"
              >
                Do students need to bring food storage containers? 
              </h5>
               
              <div
                class="px-0 container"
              >
                <div
                  class=""
                  id="bring-radio"
                  role="radiogroup"
                  tabindex="-1"
                >
                  <div
                    class="custom-control custom-radio"
                  >
                    <input
                      autocomplete="off"
                      class="custom-control-input"
                      name="bring-radio"
                      type="radio"
                      value="true"
                    />
                    <label
                      class="custom-control-label"
                    >
                      <span>
                        Yes
                      </span>
                    </label>
                  </div>
                   
                  <div
                    class="mt-1 custom-control custom-radio"
                  >
                    <input
                      autocomplete="off"
                      class="custom-control-input"
                      name="bring-radio"
                      type="radio"
                      value="false"
                    />
                    <label
                      class="custom-control-label"
                    >
                      <span>
                        No
                      </span>
                    </label>
                  </div>
                </div>
              </div>
               
              <h2
                class="mt-4 h2"
              >
                Preview
              </h2>
               
              <div
                class="custom-preview-box"
              >
                <span>
                  Hot Indian buffet food
                </span>
                 from
                
                <span>
                  FIUTS weekly club meeting
                </span>
                .
                
                <br />
                 
                <br />
                
                Quantity: 
                <span>
                  About 8 full meals
                </span>
                 
                <br />
                
                End time:
                
                <span>
                  --:-- --
                </span>
                 
                <br />
                
                Location: 
                <span>
                  HUB 130
                </span>
                 
                <!---->
                 
                <br />
                 
                <!---->
              </div>
               
              <div
                style="margin-top: 32px;"
              >
                <div
                  class="row justify-content-between"
                >
                  <div
                    class="col-md-5 col-lg-4 order-md-2"
                  >
                    <button
                      class="btn mb-3 btn-primary btn-lg btn-block"
                      style="white-space: nowrap;"
                      type="submit"
                    >
                      Submit
                    </button>
                  </div>
                   
                  <div
                    class="col-md-5 col-lg-4 order-md-1"
                  >
                    <button
                      class="btn mb-3 btn-outline-secondary btn-lg btn-block"
                      style="white-space: nowrap;"
                      type="reset"
                    >
                      Reset
                    </button>
                  </div>
                </div>
              </div>
            </form>
          </div>
           
        </div>
      </div>
    </div>
  </div>
   
  <footer
    class="text-center"
    id="relative-footer"
  >
    <a
      href="mailto:help@uw.edu?subject=Hungry Husky support"
    >
      Contact support
    </a>
     
    <p>
      Â© 2019 University of Washington
    </p>
  </footer>
</div>
`;

exports[`Food Categories Page renders correctly 1`] = `
<div
  class="page"
>
  <div
    id="standard-notification"
  >
    <div
      class="collapse"
      id="notif-container"
      style="display: none;"
    >
      <div
        class="text-center py-3 container-fluid"
      >
        <div
          class="container"
        >
          <div
            class="row justify-content-center"
          >
            <div
              class="col-md-8"
            >
              <div
                class="h5 text-white pt-2"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="page-content pb-2"
  >
     
    <div
      class="container"
    >
      <div
        class="row justify-content-center"
      >
        <div
          class="page-content-padding col-md-8"
        >
          <h1
            class="pt-4 pb-2"
            id="standard-heading"
          >
            
        Compose notification
    
          </h1>
           
          <div
            id="standard-body"
          >
            <div>
              <div
                id="submitconfirmation___BV_modal_outer_"
                style="position: absolute; z-index: 1040;"
              >
                <div
                  aria-hidden="true"
                  class="modal fade"
                  id="submitconfirmation"
                  role="dialog"
                  style="display: none;"
                  tabindex="-1"
                >
                  <div
                    class="modal-dialog modal-md"
                  >
                    <div
                      aria-describedby="submitconfirmation___BV_modal_body_"
                      aria-labelledby="submitconfirmation___BV_modal_header_"
                      class="modal-content"
                      id="submitconfirmation___BV_modal_content_"
                      role="document"
                    >
                      <header
                        class="modal-header"
                        id="submitconfirmation___BV_modal_header_"
                      >
                        <h5
                          class="modal-title"
                        >
                          Confirmation
                        </h5>
                        <button
                          aria-label="Close"
                          class="close"
                          type="button"
                        >
                          Ã—
                        </button>
                      </header>
                      <div
                        class="modal-body"
                        id="submitconfirmation___BV_modal_body_"
                      >
                        <p>
                          
                We will send your notification to Hungry Husky Subscribers.
            
                        </p>
                         
                        <div
                          class="custom-preview-box"
                        >
                          <span>
                            Hot Indian buffet food
                          </span>
                           from
                
                          <span>
                            FIUTS weekly club meeting
                          </span>
                          .
                
                          <br />
                           
                          <br />
                          
                Quantity: 
                          <span>
                            About 8 full meals
                          </span>
                           
                          <br />
                          
                End time:
                
                          <span>
                            --:-- --
                          </span>
                           
                          <br />
                          
                Location: 
                          <span>
                            HUB 130
                          </span>
                           
                          <!---->
                           
                          <br />
                           
                          <!---->
                        </div>
                      </div>
                      <footer
                        class="modal-footer"
                        id="submitconfirmation___BV_modal_footer_"
                      >
                        <button
                          class="btn btn-outline-secondary"
                          type="button"
                        >
                          Cancel
                        </button>
                        <button
                          class="btn btn-primary"
                          type="button"
                        >
                          Send
                        </button>
                      </footer>
                    </div>
                  </div>
                </div>
                <!---->
                <!---->
              </div>
            </div>
             
            <form
              class=""
            >
              <label
                class="mt-2 standard-label"
                for="event-name"
              >
                Event name
              </label>
               
              <input
                aria-describedby="Name of the event"
                aria-required="true"
                class="mb-3 standard-placeholder form-control"
                id="event-name"
                placeholder="FIUTS weekly club meeting"
                required="required"
                type="text"
              />
               
              <label
                class="standard-label"
                for="food-description"
              >
                Describe the food
              </label>
               
              <textarea
                aria-describedby="Describe the food"
                aria-required="true"
                class="mb-3 standard-placeholder form-control"
                id="food-description"
                placeholder="Hot Indian buffet food"
                required="required"
                rows="2"
                wrap="soft"
              />
               
              <label
                class="standard-label"
                for="quantity"
              >
                Quantity
              </label>
               
              <input
                aria-describedby="Quantity of food"
                aria-required="true"
                class="mb-3 standard-placeholder form-control"
                id="quantity"
                placeholder="About 8 full meals"
                required="required"
                type="text"
              />
               
              <label
                class="standard-label"
                for="end-time"
              >
                End time (when food service will end)
              </label>
               
              <input
                aria-describedby="End time of the event"
                aria-required="true"
                class="mb-3 standard-placeholder form-control"
                id="end-time"
                required="required"
                type="time"
              />
               
              <label
                class="standard-label"
                for="location"
              >
                Location
              </label>
               
              <input
                aria-describedby="Location of the event"
                aria-required="true"
                class="mb-4 standard-placeholder form-control"
                id="location"
                placeholder="HUB 130"
                required="required"
                type="text"
              />
               
              <h2
                class="mt-4"
              >
                Food specifications
              </h2>
               
              <h5
                class="standard-label"
              >
                Does the food contain allergens?
              </h5>
               
              <p>
                It's ok if you are unsure, just select to the best of your knowledge.
              </p>
               
              <div
                class="px-0 container"
              >
                <div
                  class=""
                  id="allergens-checkbox"
                  role="group"
                  tabindex="-1"
                >
                  <div
                    class="row"
                  />
                </div>
              </div>
               
              <h5
                class="mt-4 standard-label"
              >
                Do students need to bring food storage containers? 
              </h5>
               
              <div
                class="px-0 container"
              >
                <div
                  class=""
                  id="bring-radio"
                  role="radiogroup"
                  tabindex="-1"
                >
                  <div
                    class="custom-control custom-radio"
                  >
                    <input
                      autocomplete="off"
                      class="custom-control-input"
                      name="bring-radio"
                      type="radio"
                      value="true"
                    />
                    <label
                      class="custom-control-label"
                    >
                      <span>
                        Yes
                      </span>
                    </label>
                  </div>
                   
                  <div
                    class="mt-1 custom-control custom-radio"
                  >
                    <input
                      autocomplete="off"
                      class="custom-control-input"
                      name="bring-radio"
                      type="radio"
                      value="false"
                    />
                    <label
                      class="custom-control-label"
                    >
                      <span>
                        No
                      </span>
                    </label>
                  </div>
                </div>
              </div>
               
              <h2
                class="mt-4 h2"
              >
                Preview
              </h2>
               
              <div
                class="custom-preview-box"
              >
                <span>
                  Hot Indian buffet food
                </span>
                 from
                
                <span>
                  FIUTS weekly club meeting
                </span>
                .
                
                <br />
                 
                <br />
                
                Quantity: 
                <span>
                  About 8 full meals
                </span>
                 
                <br />
                
                End time:
                
                <span>
                  --:-- --
                </span>
                 
                <br />
                
                Location: 
                <span>
                  HUB 130
                </span>
                 
                <!---->
                 
                <br />
                 
                <!---->
              </div>
               
              <div
                style="margin-top: 32px;"
              >
                <div
                  class="row justify-content-between"
                >
                  <div
                    class="col-md-5 col-lg-4 order-md-2"
                  >
                    <button
                      class="btn mb-3 btn-primary btn-lg btn-block"
                      style="white-space: nowrap;"
                      type="submit"
                    >
                      Submit
                    </button>
                  </div>
                   
                  <div
                    class="col-md-5 col-lg-4 order-md-1"
                  >
                    <button
                      class="btn mb-3 btn-outline-secondary btn-lg btn-block"
                      style="white-space: nowrap;"
                      type="reset"
                    >
                      Reset
                    </button>
                  </div>
                </div>
              </div>
            </form>
          </div>
           
        </div>
      </div>
    </div>
  </div>
   
  <footer
    class="text-center"
    id="relative-footer"
  >
    <a
      href="mailto:help@uw.edu?subject=Hungry Husky support"
    >
      Contact support
    </a>
     
    <p>
      Â© 2019 University of Washington
    </p>
  </footer>
</div>
`;
